src/core/providers/meshmemory/qgsmeshmemorydataprovider.cpp

```diff
@@ -290,12 +308,38 @@ bool QgsMeshMemoryDataProvider::checkDatasetValidity( std::shared_ptr<QgsMeshMem
       setError( QgsError( tr( "Dataset defined on faces has {} values, but mesh {}" ).arg( dataset->values.count(), faceCount() ),
                           QStringLiteral( "Mesh Memory Provider" ) ) );
     }
+  } else if ( dataType == QgsMeshDatasetGroupMetadata::DataOnEdges )
+  {
+    // on edges
+    if ( dataset->values.count() != edgeCount() )
+    {
+      valid = false;
+      setError( QgsError( tr( "Dataset defined on edges has {} values, but mesh {}" ).arg( dataset->values.count(), edgeCount() ),
+                          QStringLiteral( "Mesh Memory Provider" ) ) );
+    }
   }
-
   dataset->valid = valid;
   return valid;
 }
 
+bool QgsMeshMemoryDataProvider::checkVertexId( int vertex_id )
```

Comment:
vertex_id is [1..N] or [0..N-1] ? I think in either case one of the two followng ifs will need changing to <=

Reply:
its [0..N-1], fixed <=